### ISSUE TITLE
File picker doesn't work on linux

### ISSUE THREAD

**Issue by kroune** [vinceglb/FileKit#178]
When using file picker on linux (I have used sample compose project) it spams with errors and doesn't work

![image](https://github.com/user-attachments/assets/29e0c291-78f5-4bbf-93e1-a7fbd5ed4afe)

`(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkViewport': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkPlacesSidebar': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkPaned': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkBox': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkFileChooserWidget': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkBox': error occurred in libfreetype

(java:140182): Gtk-WARNING **: 00:04:56.728: drawing failure for widget 'GtkFileChooserDialog': error occurred in libfreetype
`

System info: Arch linux
File manager: Thunar

**Comment by vinceglb** [comment]
Hi @kroune! Thanks for sharing this issue!

I found this related issue: https://youtrack.jetbrains.com/issue/CMP-6022

I will probably need some help to solve this one, it seems not trivial.

**Comment by Duron27** [comment]
funny I just got this issue today, but didn't have it lastnight. and I didn't update my linux or any file related code. I did however update some libraries in my gradle file. I also did a git push lastnight when it was working. I'll see if I can find the change

**Comment by vinceglb** [comment]
@kroune, does this solve your problem if you put this in your proguard file?

> -keep class com.sun.jna.** { _; } 
> -keep class * implements com.sun.jna._* { *; }

**Comment by kroune** [comment]
> [@kroune](https://github.com/kroune), does this solve your problem if you put this in your proguard file?
> 
> > -keep class com.sun.jna.** { _; }
> > -keep class * implements com.sun.jna._* { *; }

no, it doesn't help, it doesn't work even if I use runDistributable (which doesn't have proguard at all)

![Image](https://github.com/user-attachments/assets/50af4e3f-ea5a-4de4-ac47-23051252c0de)

**Comment by kroune** [comment]
After playing around with it I found that some elements are visible (sometimes) and it seems like it works a bit (but ui is still completely fucked)

![Image](https://github.com/user-attachments/assets/f606dd4f-c011-438a-868f-9aeace8b68a6)

Also I think it would be nice to have another implementation of a file picker as a workaround

**Comment by joshgiesbrecht** [comment]
Just FYI, we're having a similar problem with the Processing project on Linux. If we sort it out I'll let you know.
https://github.com/processing/processing4/issues/1147
(In our case we have a workaround, which is using the Java file picker instead of the native Linux OS one.)

**Comment by nicopico-dev** [comment]
> [@kroune](https://github.com/kroune), does this solve your problem if you put this in your proguard file?
> 
> > -keep class com.sun.jna.** { _; }
> > -keep class * implements com.sun.jna._* { *; }

These rules solved a crash I had on my application on MacOS. This information should be added to the documentation

**Comment by vinceglb** [comment]
@nicopico-dev I updated the JVM setup documentation with a proguard section ✅

https://filekit.mintlify.app/core/setup#proguard-configuration

**Comment by vinceglb** [comment]
I'm closing the issue for now. Feel free to reopen!
